src/Management/src/Endpoint/Mappings/MappingDescription.cs

```diff
@@ -50,16 +50,22 @@ private RouteMappingDetails CreateMappingDetails(IRouteDetails routeDetails)
                 Consumes = routeDetails.Consumes.Select(consumes => new MediaTypeDescriptor
                 {
                     MediaType = consumes
-                }).ToArray(),
+                }).ToList(),
                 Produces = routeDetails.Produces.Select(produces => new MediaTypeDescriptor
                 {
                     MediaType = produces
-                }).ToArray(),
-                Methods = routeDetails.HttpMethods.ToArray(),
+                }).ToList(),
+                Methods = routeDetails.HttpMethods.ToList(),
                 Patterns = new[]
                 {
                     routeDetails.RouteTemplate
-                }
+                },
+                Params = new string[]
+                {
+                }, // does not apply for .NET
+                Headers = new string[]
+                {
+                } // Cannot infer here for .NET 
```

Comment:
Why return a list in some cases, and an array in others? I suppose returning a list in all cases is slightly more convenient to calling code that wants to modify the collections.

Reply:
fixed to return list all the time.